tar/asm/disassemble.go

```diff
@@ -135,6 +135,9 @@ func NewInputTarStream(r io.Reader, p storage.Packer, fp storage.FilePutter) (io
 				}
 				isEOF = true
 			}
+			if n == 0 {
+				break
```

Comment:
https://pkg.go.dev/io#Reader says
>  Callers should treat a return of 0 and nil as indicating that nothing happened; in particular it does not indicate EOF.

so I think this should, pedantically, be
```go
if n != 0 {
   if _, err := AddEntry(…); err != nil { fail }
}
```
letting `isEOF` break the loop, but continuing to the next iteration on `Read` returning `0, nil`.

Reply:
Filed #68 .